unreal_asset/src/lib.rs

```diff
@@ -1568,185 +711,62 @@ impl<'a, C: Read + Seek> Asset<C> {
-                                        true => self
-                                            .get_import(struct_property.struct_value)
-                                            .map(|e| e.object_name.content.to_owned()),
-                                        false => None,
-                                    }
-                                }
-                                _ => None,
-                            };
-                            if let Some(key) = key_override {
-                                self.map_key_override
-                                    .insert(map.generic_property.name.content.to_owned(), key);
-                            }
-
-                            let value_override = match &*map.value_prop {
-                                FProperty::FStructProperty(struct_property) => {
-                                    match struct_property.struct_value.is_import() {
-                                        true => self
-                                            .get_import(struct_property.struct_value)
-                                            .map(|e| e.object_name.content.to_owned()),
-                                        false => None,
-                                    }
-                                }
-                                _ => None,
-                            };
-
-                            if let Some(value) = value_override {
-                                self.map_value_override
-                                    .insert(map.generic_property.name.content.to_owned(), value);
-                            }
-                        }
-                    }
-                    class_export.into()
-                } else if export_class_type.content.ends_with("Property") {
-                    PropertyExport::from_base(base_export, self)?.into()
-                } else {
-                    NormalExport::from_base(base_export, self)?.into()
-                }
+        if self.header_offset > 0 && !self.asset_data.exports.is_empty() {
+            let mut new_exports = Vec::with_capacity(self.asset_data.exports.len());
+            for i in 0..self.asset_data.exports.len() {
+                let export = self.read_export(i)?;
+                new_exports.push(export);
             }
-        };
 
-        let extras_len =
-            next_starting as i64 - self.cursor.stream_position().unwrap_or_default() as i64;
-        if extras_len < 0 {
-            // todo: warning?
-
-            self.cursor
-                .seek(SeekFrom::Start(base_export.serial_offset as u64))?;
-            return Ok(RawExport::from_base(base_export.clone(), self)?.into());
-        } else if let Some(normal_export) = export.get_normal_export_mut() {
-            let mut extras = vec![0u8; extras_len as usize];
-            self.cursor.read_exact(&mut extras)?;
-            normal_export.extras = extras;
+            self.asset_data.exports = new_exports;
```

Comment:
Use `Vec::retain` to avoid reallocating.

Reply:
I'm not really sure how `Vec::retain` helps here, as all the exports will be replaced with the new ones, there will be nothing to retain

Reply:
Doing something like map would require somehow getting results out to throw an error if something went wrong